### DISCUSSION TITLE
GPSD support

### DISCUSSION THREAD

**Discussion by MrSpock** [pendulum-project/ntpd-rs#1361] [Q&A] [answered]
I'm considering to migrate ntpd to ntpd-rs.
I'm unable to find information is it supporting gpsd as as clock source.
ndpd support special IP address that is triggering access to shared memory with GPSD deamon:
https://gist.github.com/edro15/c3fbaaabfe31ecb799363ffab587f336

Is it supported by ntpd-rs ?

**Comment by davidv1992** [comment, marked as answer]
Unfortunately not yet. We are planning to write this in the medium term, once we can secure funding for it.

**Reply by MrSpock** [reply]
Do you accept pull requests ? ;-)

**Reply by davidv1992** [reply]
Apologies for the delayed response. Yes we do accept pull requests.

We have internally already spent some time thinking on this issue, and we do have some quite strong preferences for direction. First of all for precision of the GPS signal, this will also require pulse per second support, which requires some modifications to the filtering code as well to deal with sources that only partially constrain the time (in the case of pulse per second only within the second). Also, we would strongly prefer not to support the shared memory approach (which is tricky to get to interact well with rusts ownership model) but rather to use the socket based communication GPSD provides for chrony.

If you are willing to take this up, I would suggest splitting the work up into three pieces, with a PR for each
1) the algorithm modifications needed for pulse-per-second like sources
2) actual pps support itself
3) the gpsd socket support
This would make review for us a bit easier, and hence faster.
